### PR TITLE
feat: Refactor modal component, Implement Delete Vault Feature

### LINKED ISSUE #1284: Delete credential removes the vault as well from the vault screen

### LINKED ISSUE #1299: Move common styles of the modal to modal.vue and refactor deleteCred and Add vault modals

### LINKED ISSUE #1307: Implement Delete Vault Feature

### PR DESCRIPTION
**Pull request by talwinder50**
closes #1299

closes #1284

closes #1307

Signed-off-by: talwinder50 <talwinderkaur50@gmail.com>
Screen Shot 2021-12-09 at 6 29 16 PM
Screen Shot 2021-12-09 at 6 30 51 PM
Screen Shot 2021-12-09 at 6 31 03 PM

https://user-images.githubusercontent.com/7862595/145494054-7c4995ac-b404-4f69-962a-82ef4a946d28.mov

### CODE REVIEW

File: cmd/wallet-web/src/components/CredentialDetails/DeleteCredentialModal.vue [hunk 1]
```diff
@@ -4,82 +4,47 @@
...
       </div>
-    </div>
+    </template>
+    <!-- Buttons Container -->
+    <template #actionButton>
+      <styled-button
+        id="deleteButton"
+        class="order-first md:order-last lg:order-last w-full md:w-auto lg:w-auto"
+        type="danger"
+        @click="deleteCredential(credentialId)"
```
Comment (sudeshrshetty) [inline cmd/wallet-web/src/components/CredentialDetails/DeleteCredentialModal.vue:43]: may not be related to this PR, but delete credential function doesn't return anything.

Still I see catching return params in code below
`const resp = await credentialManager.remove(token, credID);` inside deleteCredential function.

**Review by sudeshrshetty — APPROVED** [review]

File: cmd/wallet-web/src/components/CredentialDetails/DeleteCredentialModal.vue (hunk 1 above)
Comment (talwinder50) [reply cmd/wallet-web/src/components/CredentialDetails/DeleteCredentialModal.vue:43]: yea i have noticed i will remove that in follow up pr

File: cmd/wallet-web/src/components/CredentialDetails/DeleteCredentialModal.vue [hunk 2]
```diff
@@ -4,92 +4,58 @@
...
+          </p>
         </div>
       </div>
-    </div>
+    </template>
+    <!-- Buttons Container -->
+    <template #actionButton>
+      <styled-button
+        id="deleteButton"
+        class="order-first md:order-last lg:order-last w-full md:w-auto lg:w-auto"
```
Comment (birtony) [inline cmd/wallet-web/src/components/CredentialDetails/DeleteCredentialModal.vue:41]: You have added excessive classes for breakpoints above `md` again. Please, remember that `md` already applies to everything above this brake point, so you do not need to specify the same class for `lg`

File: cmd/wallet-web/src/components/CredentialDetails/DeleteCredentialModal.vue [hunk 3]
```diff
@@ -4,92 +4,58 @@
...
-            {{ t('CredentialDetails.deleteCredential') }}?
-          </span>
-          <div class="relative flex-auto">
-            <p class="pb-12 text-base text-center text-neutrals-medium">
-              {{ t('CredentialDetails.deleteCredentialConfirmMessage') }}
-            </p>
-          </div>
+  <modal :show="showModal">
+    <template #content>
+      <!--content-->
```
Comment (birtony) [inline cmd/wallet-web/src/components/CredentialDetails/DeleteCredentialModal.vue:9]: Can we call this `<!-- Modal Content -->` and move to the line above the `template`?

File: cmd/wallet-web/src/components/CredentialDetails/DeleteCredentialModal.vue [hunk 4]
```diff
@@ -4,92 +4,58 @@
...
+            {{ t('CredentialDetails.deleteCredentialConfirmMessage') }}
+          </p>
         </div>
       </div>
-    </div>
+    </template>
+    <!-- Buttons Container -->
+    <template #actionButton>
+      <styled-button
+        id="deleteButton"
```
Comment (birtony) [inline cmd/wallet-web/src/components/CredentialDetails/DeleteCredentialModal.vue:40]: Let's provide a more explicit `id`, maybe something like `delete-credential-button`

File: cmd/wallet-web/src/components/CredentialDetails/DeleteCredentialModal.vue [hunk 5]
```diff
@@ -4,92 +4,58 @@
...
-import { mapGetters } from 'vuex';
-import { CredentialManager } from '@trustbloc-cicd/wallet-sdk';
-import { decode } from 'js-base64';
-import { useI18n } from 'vue-i18n';
+import {ref, watch} from 'vue';
+import {mapGetters} from 'vuex';
+import {CredentialManager} from '@trustbloc-cicd/wallet-sdk';
+import {decode} from 'js-base64';
+import {useI18n} from 'vue-i18n';
+import StyledButton from '@/components/StyledButton/StyledButton';
```
Comment (birtony) [inline cmd/wallet-web/src/components/CredentialDetails/DeleteCredentialModal.vue]: Looks like you didn't run linter

**Review by rolsonquadras — APPROVED** [review]

File: cmd/wallet-web/src/components/Modal/Modal.vue [hunk 6]
```diff
@@ -4,19 +4,94 @@
...
+        inset-0
+        z-50
+        justify-center
+        items-center
+        bg-neutrals-dark bg-opacity-50
+      "
+    >
+      <div class="mx-6 lg:mx-auto flex-grow max-w-lg bg-neutrals-white rounded-2xl">
+        <slot name="content" />
+        <!-- Button Container -->
```
Comment (birtony) [inline cmd/wallet-web/src/components/Modal/Modal.vue:22]: Buttons*

File: cmd/wallet-web/src/components/Modal/Modal.vue [hunk 7]
```diff
@@ -4,19 +4,94 @@
...
+            bg-neutrals-magnolia
+            rounded-b-2xl
+            flex flex-col
+            modal-footer
+            border-t border-0 border-neutrals-lilacSoft
+          "
+        >
+          <styled-button
+            type="outline"
+            class="w-full md:w-auto lg:w-auto"
```
Comment (birtony) [inline cmd/wallet-web/src/components/Modal/Modal.vue:44]: `lg:w-auto` is, again, excessive

File: cmd/wallet-web/src/components/Vaults/AddVaultModal.vue [hunk 8]
```diff
@@ -5,104 +5,56 @@
...
+          :label="t('Vaults.addlabel')"
+          :placeholder="t('Vaults.placeholderLabel')"
+          :value="vaultName"
+          type="text"
+          maxlength="42"
+          @update="receivedVaultName($event)"
+        />
+      </div>
+    </template>
+    <!--button container-->
```
Comment (birtony) [inline cmd/wallet-web/src/components/Vaults/AddVaultModal.vue:48]: Maybe we just keep these comments in the Modal itself and remove from here, the template name is self-explanatory. Besides, can we be consistent and use camel-case for comments like this, which defines a title of the section in code? `<!-- Buttons Container -->`

File: cmd/wallet-web/src/components/Vaults/AddVaultModal.vue [hunk 9]
```diff
@@ -5,104 +5,56 @@
...
-        inset-0
-        z-50
-        justify-center
-        items-center
-        bg-neutrals-dark bg-opacity-50
-      "
-    >
+  <modal :show="showModal">
+    <template #content>
+      <!--content-->
```
Comment (birtony) [inline cmd/wallet-web/src/components/Vaults/AddVaultModal.vue:10]: Maybe we just keep these comments in the Modal itself and remove from here, the template name is self-explanatory

File: cmd/wallet-web/src/components/Vaults/DeleteVaultModal.vue [hunk 10]
```diff
@@ -0,0 +1,115 @@
+<!--
+ * Copyright SecureKey Technologies Inc. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+-->
+<template>
+  <modal :show="showModal">
+    <template #content>
+      <!--content-->
```
Comment (birtony) [inline cmd/wallet-web/src/components/Vaults/DeleteVaultModal.vue:9]: Maybe we just keep these comments in the Modal itself and remove from here, the template name is self-explanatory

File: cmd/wallet-web/src/components/Vaults/DeleteVaultModal.vue [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+          {{ t('Vaults.deleteVault') }}?
+        </span>
+        <div class="relative flex-auto">
+          <p class="pb-12 text-base text-center text-neutrals-medium">
+            {{ t('Vaults.deleteVaultConfirmMessage') }}
+          </p>
+        </div>
+      </div>
+    </template>
+    <!-- Buttons Container -->
```
Comment (birtony) [inline cmd/wallet-web/src/components/Vaults/DeleteVaultModal.vue:37]: Maybe we just keep these comments in the Modal itself and remove from here, the template name is self-explanatory

File: cmd/wallet-web/src/pages/Credentials.vue [hunk 12]
```diff
@@ -123,10 +123,6 @@
     <!-- Loading State -->
     <skeleton-loader v-if="loading" type="vault" />
     <!-- Error State -->
-    <span v-else-if="loadingStatus === 'failed'">
-      <b>Warning:</b> Failed to connect to server. Your wallet can not participate in secured
-      communication.
-    </span>
```
Comment (birtony) [inline cmd/wallet-web/src/pages/Credentials.vue:129]: We should at least console error it if the loading status is failed

File: cmd/wallet-web/src/translations/en.js [hunk 13]
```diff
@@ -8,6 +8,7 @@ import Signin from './Signin/en.json';
 import Signout from './Signout/en.json';
 import Signup from './Signup/en.json';
 import Vaults from './Vaults/en.json';
+import Modal from './Modal/en.json';
```
Comment (birtony) [inline cmd/wallet-web/src/translations/en.js:11]: Can we add components in alphabetical order here?

File: cmd/wallet-web/src/translations/en.js [hunk 14]
```diff
@@ -20,4 +21,5 @@ export default {
   Signout,
   Signup,
   Vaults,
+  Modal,
```
Comment (birtony) [inline cmd/wallet-web/src/translations/en.js:24]: And here

**Review by birtony — CHANGES_REQUESTED** [review]
This modal implementation is very nice, left some minor comments

File: cmd/wallet-web/src/pages/Credentials.vue (hunk 12 above)
Comment (talwinder50) [reply cmd/wallet-web/src/pages/Credentials.vue]: this should not have been deleted, this was for my local env only. not pushing this change.

**Review by birtony — APPROVED** [review]
We can merge this and address outstanding issues, which were filed. We need to revisit the login on how we show/close modal, to avoid having it closed if you click somewhere on the modal itself

**Pull request merged**